lib/loadConfig.js

```diff
@@ -42,7 +47,33 @@ const jsonParse = (path, content) => {
   }
 }
 
-const yamlParse = (path, content) => YAML.parse(content)
+const json5Parse = (path, content) => {
+  try {
+    const json5 = JSON5.parse(content)
+    return json5?.['lint-staged'] ?? json5
+  } catch (error) {
+    if (path.endsWith(PACKAGE_JSON5)) {
+      debugLog('Ignoring invalid package file `%s` with content:\n%s', path, content)
+      return undefined
+    }
+
+    throw error
+  }
+}
+
+const yamlParse = (path, content) => {
+  try {
+    const yaml = YAML.parse(content)
+    return yaml?.['lint-staged'] ?? yaml
```

Comment:
This seems to be adding extra logic to the current loader; I ewould rather avoid it.

Can you tell me why you need to support loading config from these new package formats? Wouldn't using a separate lint-staged config work?

The reason for not really liking this idea is that it's nonstandard compared to `npm` and I would like to keep things simple.

Reply:
So, let's limit the logic of using `lint-staged` key for when the filename is exactly `package.yaml` or `package.yml`.

Reply:
> Can you tell me why you need to support loading config from these new package formats? Wouldn't using a separate lint-staged config work?

Being _forced_ to create a seperate config file containing maybe just a few lines, for every project is far from ideal…

By this logic, why then support reading config from `package.json`?

> This seems to be adding extra logic to the current loader; I would rather avoid it.

As you mentioned in https://github.com/lint-staged/lint-staged/issues/1343#issuecomment-1783077629:

> it probably assumes the file is directly the config, instead of there being the `lint-staged` key inside it.

This is why this approach was added. If you can think of a better way—as someone more familiar with this codebase—@iiroj, I’m open to suggestions…

> The reason for not really liking this idea is that it's nonstandard compared to `npm`

As mentioned in #1344, it is [fully supported](https://github.com/pnpm/pnpm/issues/1803) in `pnpm` now, which has been growing in popularity as a project for some time…

> I would like to keep things simple.

I’m all for simplicity! But it’s a balance between package maintenance, and forcing tricky workarounds and/or extra config/files on users…

Also, removing JSON5 now substantially cuts down the additions here.

Reply:
> So, let's limit the logic of using `lint-staged` key for when the filename is exactly `package.yaml` or `package.yml`.

@iiroj Done.